src/pages/SetNewPassword.tsx

```diff
@@ -0,0 +1,217 @@
+import { useEffect, useState } from "react";
+import {
+  Heading,
+  Input,
+  Button,
+  Notification,
+  Link,
+} from "@stellar/design-system";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { USE_SSO, LOCAL_STORAGE_SESSION_TOKEN } from "constants/settings";
+import { singleUserStore } from "helpers/singleSingOn";
+
+import { AppDispatch, resetStoreAction } from "store";
+import { setNewPasswordAction } from "store/ducks/forgotPassword";
+import { useRedux } from "hooks/useRedux";
+
+export const SetNewPassword = () => {
+  const dispatch: AppDispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const { forgotPassword } = useRedux("forgotPassword");
+
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [confirmNewPassword, setConfirmNewPassword] = useState("");
+
+  const [errorPassword, setErrorPassword] = useState("");
+  const [errorNewPassword, setErrorNewPassword] = useState("");
+  const [errorPasswordMatch, setErrorPasswordMatch] = useState("");
+
+  useEffect(() => {
+    if (forgotPassword.status === "SUCCESS") {
+      setCurrentPassword("");
+      setNewPassword("");
+      setConfirmNewPassword("");
+    }
+  }, [forgotPassword.status]);
+
+  const handleSignOut = () => {
+    if (USE_SSO) {
+      // reset user store (from session storage)
+      singleUserStore().then();
+    }
+    dispatch(resetStoreAction());
+    localStorage.removeItem(LOCAL_STORAGE_SESSION_TOKEN);
+  };
+
+  const goToSignIn = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    handleSignOut();
+    navigate("/");
+  };
+
+  const validatePassword = () => {
+    setErrorPassword(currentPassword ? "" : "Current password is required");
+  };
+
+  const validateNewPassword = () => {
```

Comment:
this looks pretty similar to the password validation in `src/pages/ResetPassword.tsx`. It'd be great to move this logic into a helper to prevent duplication, if possible